src/plugins/OauthPlugin/OauthPlugin.js

```diff
@@ -61,43 +83,74 @@ export default class OauthPlugin extends PluginBase {
     this.oauthRequestProvider = new Oauth({
       clientId: this.oauthClientId,
       scopes: options.scopes,
-      oauthPopup: options.oauthPopup,
       oauthServer: this.oauthServer,
       oauthStrategy,
       frameStrategy: this.frameStrategy,
     });
   }
 
+  /**
+   * @param {string} [source]
+   * 
+   * @returns {boolean}
+   */
   isSourceEqualTarget(source) {
+    // @ts-ignore
+    // Here we have excellent example of duck-overtyped logic.
+    // Seems to we compare string with WIndow object (or null),
+    // and if so, this predicate will return `true` only if source
+    // will be `null`. For now will better ignore it.
     return source === this.frameStrategy.target;
```

Comment:
when target can be string? or source can be string?

Reply:
Where we provide this target somewhere. From code, unfortunately, impossible to detect type with 100% precise, so, based by existed information, I did - now I see, incorrect - assertion.

Reply:
Fixed, did declare new type `ContextWindow` for that.